types/tests/package.json

```diff
@@ -0,0 +1,27 @@
+{
```

Comment:
I don't know how the typescript types are generated, to be honest. 

@elektronik2k5 If it's possible to use only one `package.json`, then I think it would be way better.

Reply:
In JS projects, TS types aren't generated, but are written by hand. I wrote the types in `types/react-permissible.d.ts`. This package.json file has nothing to do with writing types.

The idea of the setup is meant to mitigate a bug, which I previously solved in another [repo](https://github.com/caiogondim/fast-memoize.js/pull/69). Instead of testing types against your package source, which gets transformed before publishing, I'm testing the types against the built package using `npm pack`. This has two benefits:
1. We test against what's actually published (without publishing) - so we know it works with the real package and not just source.
2. Dependencies which are used **just for testing** the types aren't included in your package, but are scoped to the small, separate test package.

So this `types/tests/package.json` is exactly that:
1. Dependencies used only for testing TS types, which aren't even installed when a user installs the main package
2. Scripts which test TS types

I hope that clarifies it :smiley:. Let me know if you have any questions or a better suggestion on how to implement these tests.

Reply:
That's a great explanation here. After reconsidering, I think it's not a problem to have a separate `package.json` for TS types. 

Tbh, I don't have much experience with TS types, let's say I'm the one learning here 😎